metricq/history_client.py

```diff
@@ -668,3 +706,49 @@ async def _on_history_response(self, message: aio_pika.IncomingMessage):
             except HistoryError as e:
                 logger.debug("message is a history response containing an error: {}", e)
                 future.set_exception(e)
+
+    def _on_history_connection_close(self, sender, _exception: Optional[Exception]):
+        self._history_connection_watchdog.set_closed()
+
+    def _on_history_connection_reconnect(self, sender, connection):
+        logger.info("History connection ({}) reestablished!", connection)
+
+        if self._reregister_task is not None and not self._reregister_task.done():
+            logger.warning(
+                "History connection was reestablished, but another reregister task is still running!"
+            )
+            self._reregister_task.cancel()
+
+        self._reregister_task = self.event_loop.create_task(
+            self._reregister(connection)
+        )
+
+        def reregister_done(task: Task):
+            try:
+                exception = task.exception()
+                if exception is None:
+                    self._history_connection_watchdog.set_established()
+                else:
+                    logger.error(
+                        f"Resubscription failed with an unhandled exception: {exception}"
+                    )
+                    raise exception
+            except CancelledError:
+                logger.warning("Resubscribe task was cancelled!")
+
+        self._reregister_task.add_done_callback(reregister_done)
+
+    async def _reregister(self, connection):
+        logger.info(
+            "Reregistering as history client...",
+        )
+        response = await self.rpc("history.register")
+        await self._declare_history_queue(response["historyQueue"])
```

Comment:
I know we do this in the sink as well, but is it generally a good idea to just ignore everything else in the the `response`? Shouldn't we at least make sure the remaining stuff hasn't changed to avoid that the manager is wrongly assuming the state of the client?